doc/source/user_documentation.rst

```diff
@@ -535,6 +531,105 @@ Here is an example of connectivity rules:
+
+	The behavior above is due to the ``distributeSynsUniformly`` flag, which is True by default. Alternatively, if it is set to False (and ``connRandomSecFromList`` is True, which is default value), a random section will be picked from the sections list for each synapse. The ``loc`` value should also be a list, and the values will be picked from it randomly and independently from section choice. If the length of sections or locations list is greater or equal to ``synsPerConn``, random choice is guaranteed to be without replacement. If ``loc`` is omitted, the value for each synapse is randomly sampled from uniform[0, 1]. 
+
+	To enforce a deterministic way of picking ``sec`` and ``loc``, set ``connRandomSecFromList`` to False (N-th synapse then gets N-th section and N-th loc from their respective lists, or if ``loc`` is a single value, it is used for all synapses). Make sure that lists of sections and locations both have the length equal to ``synsPerConn``.
+
+	If ``synsPerConn`` == 1, and a list of sections is specified, synapses (one per presynaptic cell) will be placed in sections randomly selected from the list. If ``loc`` is also a list, locations will be picked randomly (note that the random section and location will go hand in hand, i.e., the same random index is used for both).
+
+* Setting ``synMech`` as a list
+	If ``synMech`` is a list, then for each mechanism in the list, a synapse will be created. The weights, delays, and locs can be specified as lists of the same length as ``synMech``, or a single value can be used for all.
+
+	.. code-block:: python
+
+		netParams.connParams['PYR->PYR'] = {
+			'preConds': {'pop': 'src'}, 'postConds': {'pop': 'dst'},
+			'synMech': ['AMPA', 'GABA'],
+			'weight': [0.1, 0.2], # individual weights for each synapse
+			'delay': 0.1, # single delay for all synapses
+		}
+		# This will create 2 synapses: one with AMPA synMech and weight 0.1
+		# and another with GABA synMech and weight 0.2. Both will have the same delay of 0.1.
+
+
+	However, for the sections, no such one-to-one correspondence to ``synMech`` elements applies. Instead, contents of ``secs`` are repeated for each synapse.
+
+	.. code-block:: python
+
+		netParams.connParams['PYR->PYR'] = {
+			'preConds': {'pop': 'src'}, 'postConds': {'pop': 'dst'},
+			'synMech': ['AMPA', 'GABA'],
+			'sec': ['soma', 'dend']
+		}
+		# Both AMPA and GABA synapses will span 'soma' and 'dend' sections
+
+	If you want to have different sections, separate connectivity rules (``connParams`` entries) should be defined for each synaptic mechanism.
+
+* Both ``synMech`` as a list and ``synsPerConn`` > 1
+	If you have both a N-element list of ``synMechs`` and ``synsPerConn`` > 1, weights and delays can still be specified as a single value for all synapses, as a list of length N (each value corresponding to ``synMech`` list index), or a 2D list with outer dimension N (corresponding to ``synMechs``) and inner dimension corresponding to ``synsPerConn``.
```

Comment:
a 2D list with outer dimension N corresponding to ``synMechs`` and inner dimension corresponding to ``synsPerConn``.